server/src/main/java/org/elasticsearch/index/mapper/BaseGeoShapeFieldMapper.java

```diff
@@ -0,0 +1,342 @@
+            this.orientation = orientation;
+            return this;
+        }
+
+        protected Explicit<Orientation> orientation() {
+            if (orientation != null) {
+                return new Explicit<>(orientation, true);
+            }
+            return Defaults.ORIENTATION;
+        }
+
+        @Override
+        protected boolean defaultDocValues(Version indexCreated) {
+            return false;
+        }
+
+        public Builder ignoreMalformed(boolean ignoreMalformed) {
+            this.ignoreMalformed = ignoreMalformed;
+            return this;
+        }
+
+        protected Explicit<Boolean> ignoreMalformed(BuilderContext context) {
+            if (ignoreMalformed != null) {
+                return new Explicit<>(ignoreMalformed, true);
+            }
+            if (context.indexSettings() != null) {
+                return new Explicit<>(IGNORE_MALFORMED_SETTING.get(context.indexSettings()), false);
+            }
+            return Defaults.IGNORE_MALFORMED;
+        }
+
+        protected Explicit<Boolean> ignoreZValue() {
+            if (ignoreZValue != null) {
+                return new Explicit<>(ignoreZValue, true);
+            }
+            return Defaults.IGNORE_Z_VALUE;
+        }
+
+        public Builder ignoreZValue(final boolean ignoreZValue) {
+            this.ignoreZValue = ignoreZValue;
+            return this;
+        }
+
+        @Override
+        protected void setupFieldType(BuilderContext context) {
+            super.setupFieldType(context);
+
+            // field mapper handles this at build time
+            // but prefix tree strategies require a name, so throw a similar exception
+            if (name().isEmpty()) {
+                throw new IllegalArgumentException("name cannot be empty string");
+            }
+
+            BaseGeoShapeFieldType ft = (BaseGeoShapeFieldType)fieldType();
+            ft.setOrientation(orientation);
+        }
+    }
+
+    public static class TypeParser implements Mapper.TypeParser {
+
+        @Override
+        public Mapper.Builder parse(String name, Map<String, Object> node, ParserContext parserContext) throws MapperParsingException {
+            boolean coerce = Defaults.COERCE.value();
+            boolean ignoreZ = Defaults.IGNORE_Z_VALUE.value();
+            boolean ignoreMalformed = Defaults.IGNORE_MALFORMED.value();
+            Orientation orientation = Defaults.ORIENTATION.value();
+            DeprecatedParameters deprecatedParameters = new DeprecatedParameters();
+            boolean parsedDeprecatedParams = false;
+            for (Iterator<Map.Entry<String, Object>> iterator = node.entrySet().iterator(); iterator.hasNext();) {
+                Map.Entry<String, Object> entry = iterator.next();
+                String fieldName = entry.getKey();
+                Object fieldNode = entry.getValue();
+                if (DeprecatedParameters.parse(name, fieldName, fieldNode, deprecatedParameters)) {
```

Comment:
do we want to disallow them on 7.0+ indices? I'm happy with it being done in a follow-up PR, just asking.

Reply:
:+1:  Yes. I was going to do this in a separate PR to simplify backporting these changes to 6.6